addons/website_sale/models/product_image.py

```diff
@@ -41,3 +41,9 @@ def _check_valid_video_url(self):
         for image in self:
             if image.video_url and not image.embed_code:
                 raise ValidationError(_("Provided video URL for '%s' is not valid. Please enter a valid video URL.") % image.name)
+
+    def create(self, vals):
+        new_context = dict(self.env.context).copy()
+        if 'product_variant_id' in vals[0] and 'default_product_tmpl_id' in self.env.context:
```

Comment:
This is a bit tricky, and accessing `vals[0]` is dangerous if none are given (which doesn't make sense functionally but the parent `create` does handle it).

Ideally it should only be done for `vals` in `vals_list` that do have `product_variant_id` set, and not done for those that do not have it set, calling super twice with different values and context... Not sure if we want/need that complexity.
If not, at least instead of using `[0]`, we should iterate them with `any()`.

Also should use `vals.get('product_variant_id')` instead of `'product_variant_id' in vals` because if it is set but empty, we probably want to keep the default context.